crates/llama-cpp-bindings/src/engine.cc

```diff
@@ -107,6 +107,33 @@ std::string string_format(const std::string& format, Args ... args)
 	return std::string(buf.get(), buf.get() + size - 1); // We don't want the '\0' inside
 }
 
+void sorted_insert_in_capacity(std::vector<size_t>& max_vec, size_t num_index, const float* nums) {
```

Comment:
guess you might wanna use https://en.cppreference.com/w/cpp/algorithm/partial_sort